docker/irods/ubuntu/18.04/Dockerfile

```diff
@@ -1,39 +1,49 @@
 ARG DOCKER_TAG_PREFIX=wsinpg
-
 FROM $DOCKER_TAG_PREFIX/ub-18.04-base:latest
 
-ARG IRODS_VERSION=4.3.0-1~bionic
-ARG IRODS_PACKAGES_URL=https://packages.irods.org
-ARG UBUNTU_RELEASE=bionic
+ARG IRODS_VERSION="4.3.0"
 
 WORKDIR /opt/docker/irods
 
 COPY ./scripts/*.sh ./scripts/
 COPY ./config/* ./config/
 COPY ./patches/* ./patches/
 
-RUN apt-get update && \
+RUN echo "debconf debconf/frontend select Noninteractive" | debconf-set-selections && \
+    apt-get update && \
     apt-get install -q -y --no-install-recommends \
-    apt-transport-https \
+    apt-utils \
     ca-certificates \
+    curl \
     gpg \
     gpg-agent \
-    curl && \
-    curl -sSL $IRODS_PACKAGES_URL/irods-signing-key.asc | apt-key add - && \
-    echo "deb [arch=amd64] $IRODS_PACKAGES_URL/apt $UBUNTU_RELEASE main" | \
-    tee /etc/apt/sources.list.d/renci-irods.list && \
-    apt-get update && \
-    apt-get install -q -y --no-install-recommends \
-    postgresql \
+    lsb-release \
     jq \
+    netcat \
     patch \
+    postgresql \
     rsyslog \
-    irods-server="$IRODS_VERSION" \
-    irods-runtime="$IRODS_VERSION" \
-    irods-database-plugin-postgres="$IRODS_VERSION" \
-    irods-icommands="$IRODS_VERSION" \
-    irods-dev="$IRODS_VERSION" && \
+    unattended-upgrades \
+    locales && \
+    locale-gen en_GB en_GB.UTF-8 && \
+    localedef -i en_GB -c -f UTF-8 -A /usr/share/locale/locale.alias en_GB.UTF-8
+
+ENV LANG=en_GB.UTF-8 \
+    LANGUAGE=en_GB \
+    LC_ALL=en_GB.UTF-8
+
+RUN curl -sSL https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://packages.irods.org/apt $(lsb_release -sc) main" | \
+    tee /etc/apt/sources.list.d/renci-irods.list && \
+    apt-get update && \
+    apt-get install -q -y --no-install-recommends \
+    irods-server="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-runtime="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-database-plugin-postgres="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-icommands="${IRODS_VERSION}-1~$(lsb_release -sc)" \
+    irods-dev="${IRODS_VERSION}-1~$(lsb_release -sc)" && \
     apt-get install -q -y -f && \
+    unattended-upgrade -d -v && \
```

Comment:
Do you think it is worth keeping the -(d)ebug option? Also I'm not sure about what people think of unattended upgrades in container. I see why you want to upgrade on building but will it work ok to keep it? I may be missing something but I think you are keeping it as part of the final container.

Reply:
That's an oversight - it should be uninstalled and the debug option is not necessary. I'll clean up all the Dockerfiles with this in.